app/models/video.rb

```diff
@@ -64,13 +64,14 @@ def fill_missing_fields
 
     begin
       youtube_values = YoutubeReader::parse_video(self.youtube_id)
-    rescue
+    rescue SocketError, NoMethodError
+      raise ArgumentError, "Invalid Youtube ID"
```

Comment:
You'll want to add punctuation to your errors since these messages get rendered directly into the errors bucket.